examples/machine_learning/geneticalgorithm.cpp

```diff
@@ -0,0 +1,181 @@
+    mutanty = mutanty ^ (pow2((randu(nsamples / 2, u32) % bits).as(f32))).as(u32);
+
+    samplex = join(0, samplex, mutantx);
+    sampley = join(0, sampley, mutanty);
+
+    //Update the value of each sample with the new coordinates
+    samplez = update(searchSpace, samplex, sampley, nsamples, n);
+}
+
+void init_samples(array& searchSpace, array& samplex, array& sampley, array& samplez, const int nsamples, const int n) {
+    setSeed(time(NULL));
+    samplex = randu(nsamples, u32) % n;
+    sampley = randu(nsamples, u32) % n;
+    samplez = update(searchSpace, samplex, sampley, nsamples, n);
+}
+
+void init(array& searchSpace, array& searchSpace_x_display, array& searchSpace_y_display, array& samplex, array& sampley, array& samplez, const int nsamples, const int n) {
+    //initialize space
+    searchSpace = range(dim4(n/2, n/2), 0) + range(dim4(n/2, n/2), 1);
+
+    searchSpace = join(0, searchSpace, flip(searchSpace, 0));
+    searchSpace = join(1, searchSpace, flip(searchSpace, 1));
+
+    //initialize display data
+    searchSpace_x_display = iota(dim4(n, 1), dim4(1, n));
+    searchSpace_y_display = iota(dim4(1, n), dim4(n, 1));
+
+    //initalize searchers
+    init_samples(searchSpace, samplex, sampley, samplez, nsamples, n);
+}
+
+void geneticSearch(bool console, const int nsamples, const int n) {
+
+    af::info();
+
+    array searchspace_x_display = 0;
+    array searchspace_y_display = 0;
+    array searchspace;
+    array samplex;
+    array sampley;
+    array samplez;
+
+    init(searchspace, searchspace_x_display, searchspace_y_display,
+            samplex, sampley, samplez, nsamples, n);
+
+    float truemax = max<float>(searchspace);
+    bool found_max = false;
+
+    if (!console) {
+        af::Window win(1600, 800, "arrayfire genetic algorithm search demo");
+        win.grid(1, 2);
+
+        do{
+            float maximum = max<float>(samplez);
+            array where_m = where(samplez == maximum);
+
+            if (maximum < truemax * 0.99) {
+                reproduce(searchspace, samplex, sampley, samplez, nsamples, n);
+                printf("current max at (%d,%d): %f (truemax %f)\n",
+                        samplex(where_m).scalar<unsigned int>(),
+                        sampley(where_m).scalar<unsigned int>(), maximum, truemax);
+            } else if (!found_max) {
+                found_max = true;
+                printf("max found at (%d,%d): %f (truemax %f)\n",
+                        samplex(where_m).scalar<unsigned int>(),
+                        sampley(where_m).scalar<unsigned int>(), maximum, truemax);
+            }
+            win(0,0).setAxesTitles("idx", "idy", "search space");
+            win(0,1).setAxesTitles("idx", "idy", "search space");
+            win(0,0).surface(searchspace_x_display, searchspace_y_display, searchspace);
+            win(0,1).scatter(samplex.as(f32), sampley.as(f32), samplez.as(f32));
+            win.show();
```

Comment:
add `if (!console)` around these. Also add a break condition when window is closed.